### PR TITLE
feat: token and tokenizer

### PR DESCRIPTION
**Pull request by JKMarkowski**
## Related Issue
Closes SAP/fundamental-styles#455

## Description
There is token following fiori3 design and new component called tokenizer

## Breaking changes
* Right now the close icon is not pseudo element. It is used as a normal button with class `fd-token__close`
```html
    <button class="fd-token__close"></button>
```

* The text should be wrapped in element with `fd-token__text` class.

### Before:
image

### After:
image
image
image
image
image

### CODE REVIEW

**Comment by JKMarkowski** [comment]
There was colors requested:
```
lighten(var(--sapButton_Hover_BorderColor), 30)
darken(var(--sapButton_BorderColor), 24)
darken(var(--sapButton_Background), 2)
```

**Comment by InnaAtanasova** [comment]
Screen Shot 2019-12-09 at 11 43 04 AM

The space between the token and the input text should be the same as the space between the tokens (0.3125rem)

**Comment by InnaAtanasova** [comment]
It's missing the Readonly state

File: src/token.scss [hunk 1]
```diff
@@ -2,66 +2,89 @@
...
   border-radius: $fd-border-radius;
-  border-width: 1px;
-  border-style: solid;
+  height: 1.625rem;
+  border: 0.0625rem solid;
+  border-color: $fd-token-border-color;
 
-  @include fd-var-color("border-color", $fd-token-border-color, --fd-token-border-color);
+  &__close {
+    @include fd-reset();
```
[the remark this answers is not in the document]
Comment (JKMarkowski) [reply src/token.scss]: Done

File: src/token.scss [hunk 2]
```diff
@@ -2,66 +2,89 @@
...
-  @include fd-var-color("background-color", $fd-token-background-color, --fd-token-background-color);
 
-  line-height: fd-space(6);
-  vertical-align: middle;
-  display: inline-block;
-  padding-left: fd-space(2);
-  padding-right: fd-space(2);
+  background-color: $fd-token-background-color;
+  color: $fd-token-color;
+  font-size: fd-font-size("0");
```
[the remark this answers is not in the document]
Comment (JKMarkowski) [reply src/token.scss]: Done

File: src/token.scss [hunk 3]
```diff
@@ -2,66 +2,89 @@
...
+    height: $fd-token-compact-height;
 
-      margin-left: fd-space(tiny);
-      vertical-align: bottom;
-      line-height: fd-space(6);
-      cursor: pointer;
+    .#{$block}__close {
+      padding: $fd-token-compact-padding-vertical $fd-token-compact-padding-horizontal;
+      vertical-align: baseline;
+      width: 1.25rem;
```
[the remark this answers is not in the document]
Comment (JKMarkowski) [reply src/token.scss]: Done

File: src/token.scss [hunk 4]
```diff
@@ -2,66 +2,89 @@
...
+  display: inline-flex;
+  align-items: center;
+  padding-top: $fd-token-padding-vertical;
+  padding-left: 0.3125rem;
+  padding-bottom: $fd-token-padding-vertical;
   border-radius: $fd-border-radius;
-  border-width: 1px;
-  border-style: solid;
+  height: 1.625rem;
+  border: 0.0625rem solid;
```
[the remark this answers is not in the document]
Comment (JKMarkowski) [reply src/token.scss]: Done

**Comment by JKMarkowski** [comment]
Every comments has been solved, the PR is ready for check.

**Comment by JKMarkowski** [comment]
All of the comments are solved, also there is text overflow prevent.

**Comment by stefanoScalzo** [comment]
Screen Shot 2020-01-09 at 9 43 56 AM
Screen Shot 2020-01-09 at 9 44 34 AM
The tokenizer is weird with overflow. Is this how we want them to be displayed with a long token?

**Comment by stefanoScalzo** [comment]
IE11 tokenizer does not work properly

**Comment by JKMarkowski** [comment]
Now token and tokenizer works properly on IE11, I had to add `float: right`, instead of `flex end`.

**Comment by InnaAtanasova** [comment]
Screen Shot 2020-01-10 at 2 53 52 PM

Please check Token Truncation Behavior section in the specs. According to this section "Tokens Longer than the available space stop at the min input and are truncated. The X for cancel is always visible."
There is a 3rem min width for the input

**Comment by InnaAtanasova** [comment]
Screen Shot 2020-01-10 at 2 54 13 PM

There should be a 0.3125rem padding on the right side for the input area. This will prevent the text "touching" the border

**Comment by InnaAtanasova** [comment]
The Tokens should be focusable (also missing keyboard support)

**Comment by JKMarkowski** [comment]
Hi @InnaAtanasova Thanks for comments, I finally fixed overflow 
Opera:
image

IE11:
image

Also focus is fixed there.

**Comment by InnaAtanasova** [comment]
> Hi @InnaAtanasova Thanks for comments, I finally fixed overflow
> Tokenizer Opera:
> image
> 
> Token IE11:
> image
> 
> Also focus is fixed there.

This is perfect! Great job!

**Comment by InnaAtanasova** [comment]
Just a question: when I focus an element (fd-token) I need to tab twice to go to the next token (I guess the second tab goes to the button?) Is it supposed to be this way?
And I am able to move to the next token only by pressing the `tab` key, not with the arrow keys?

**Comment by JKMarkowski** [comment]
Hello @InnaAtanasova The second tab goes to `button` as you said, we can prevent from it, by adding `tabindex="-1"`. About the arrow keys, it's unfortunately not possible without js. Only some element supports keyboard by default on browser, like `radio/checkbox` elements.

**Pull request merged**